Add circuitpython-as7343 to Community Bundle

joepardue opened the pull request:
This pull request adds my CircuitPython driver for the AMS AS7343 spectral sensor.

- Submodule added under `libraries/drivers/circuitpython-as7343`
- Entry added to `circuitpython_community_library_list.md`

Library repository: https://github.com/joepardue/circuitpython-as7343

Thanks for reviewing!

FoamyGuy commented on libraries/drivers/libraries/drivers/circuitpython-as7343/.gitmodules:
```diff
@@ -0,0 +1 @@
+
```
I think this part of the `git add` didn't work properly or something. It's showing just an empty .gitmodules file as the contents of the new repo. Ordinarily in the github interface this would show a link out to the other repo and a hash of the current commit: 
![image](https://github.com/user-attachments/assets/66e0f5d7-9e8d-43e4-bfd3-79d3f612f91b)

I'm not sure of the exact right way to fix it. I suspect it will be something like deleting this empty `.gitmodules` file and then updating submodules in this branch that has the new library added, that should fetch the contents of the new library which could then be commit / pushed to this branch I think.

I'll try to tinker with this a little later and see if I can figure out the way to get it corrected with more certainty.

FoamyGuy reviewed (changes requested):
Thanks for submitting this new driver. I think we need a few more things before it's ready to merge.

- The libraries that are added need to have at least 1 release made. The page here details the process to do that: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/releasing-on-github
- I think something is off with the added submodule, I've noted down in the code with a separate comment some more details
- I'm not certain if this one will matter, but I noticed the actions for your library don't pass currently: https://github.com/joepardue/circuitpython-as7343/actions and that you don't have the release actions that are typically used for circuitpython libraries. I don't know off the top of my head whether that will cause the bundling process to fail when it tries to build. If so then the easiest way to get all of the requisite files is to use cookiecutter as described on this page: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/creating-a-library to generate a project template which will include all of the necessary infrastructure files.

joepardue commented:
Foamyguy,

Thanks so much for the help. This is my first time trying to do this with
circuitpython.org bundle procedure. I'll read the suggested links and try
to get this better nailed down this weekend. I owe much to Adafruit and
wanted to repay some by creating the bundle for the community.

Joe

On Thu, May 8, 2025 at 12:59 PM foamyguy ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Thanks for submitting this new driver. I think we need a few more things
> before it's ready to merge.
>
>    - The libraries that are added need to have at least 1 release made.
>    The page here details the process to do that:
>    https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/releasing-on-github
>    - I think something is off with the added submodule, I've noted down
>    in the code with a separate comment some more details
>    - I'm not certain if this one will matter, but I noticed the actions
>    for your library don't pass currently:
>    https://github.com/joepardue/circuitpython-as7343/actions and that you
>    don't have the release actions that are typically used for circuitpython
>    libraries. I don't know off the top of my head whether that will cause the
>    bundling process to fail when it tries to build. If so then the easiest way
>    to get all of the requisite files is to use cookiecutter as described on
>    this page:
>    https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/creating-a-library
>    to generate a project template which will include all of the necessary
>    infrastructure files.
>
> ------------------------------
>
> In libraries/drivers/libraries/drivers/circuitpython-as7343/.gitmodules
> <https://github.com/adafruit/CircuitPython_Community_Bundle/pull/254#discussion_r2080111009>
> :
>
> > @@ -0,0 +1 @@
> +
>
> I think this part of the git add didn't work properly or something. It's
> showing just an empty .gitmodules file as the contents of the new repo.
> Ordinarily in the github interface this would show a link out to the other
> repo and a hash of the current commit:
> image.png (view on web)
> <https://github.com/user-attachments/assets/66e0f5d7-9e8d-43e4-bfd3-79d3f612f91b>
>
> I'm not sure of the exact right way to fix it. I suspect it will be
> something like deleting this empty .gitmodules file and then updating
> submodules in this branch that has the new library added, that should fetch
> the contents of the new library which could then be commit / pushed to this
> branch I think.
>
> I'll try to tinker with this a little later and see if I can figure out
> the way to get it corrected with more certainty.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/adafruit/CircuitPython_Community_Bundle/pull/254#pullrequestreview-2825797962>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGJBELVL47L6GUPSN6IAY4L25OEN5AVCNFSM6AAAAAB4W5TGQCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQMRVG44TOOJWGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> .com>
>

joepardue commented:
I went down a two day rabbit hole that generated 65 failed builds. I'm going to back up and carefully restart after rereading the docs. This can't be this hard!

joepardue commented:
I have struggled with this. The name is not in the format normally used, but I killed the whole thing several times trying to fix that. Also, the README files are not what I want, I have much more detailed ones, but I want to see if this thing will actually go through now. I had 65 Build Cl rejections last time so I am totally fried and fearful of what will happen now. If this passes, then I'll try to update the README and if that works, I'll try to get the name in conformity with the rest of the library files.

FoamyGuy commented:
Thanks for working through the actions checks! This is very close to ready. 

We don't ordinarily want 3rd party community libraries to include "adafruit" in the name which this package does inside of your repo. I've submitted a PR over on your repo: https://github.com/joepardue/circuitpython-as7343/pull/1 to rename the package.

I think if you merge that PR (or make your own commits with the same changes) and then make a new release for your library this should be good to go.

joepardue commented:
I've fixed the build issue that was reported:

1. The syntax error in pyproject.toml has been corrected (there was a missing newline after [build-system])
2. Created a new release (v1.0.3) with the fix
3. Updated the submodule in my fork to point to the latest release

The library should now build correctly. Please let me know if there are any other issues that need to be addressed.

FoamyGuy commented:
@joepardue thanks again for working through those issues. 

One more thing I just noticed is that there are extra nested directores with a leftover empty gitmodules file in it at:
```
libraries/drivers/libraries/drivers/circuitpython-as7343/.gitmodule
```

I think you should be able to fix that by running this command inside your CircuitPython_Community_Bundle repo:
```
git rm -r libraries/drivers/libraries/drivers/circuitpython-as7343/
```
after that make a commit and push it to this branch and I think it should be fixed. 

When you make PRs there should be an "allow maintainers to add commits" option or something similar that if you have allowed it can let the project maintainers add commits to fix things like this if you are comfortable with that. I tried to commit this fix but it was not allowing me.

FoamyGuy commented:
@joepardue 
I think the versions issue that the actions failure has in the log is due to the way the releases on the library are:
![image](https://github.com/user-attachments/assets/c4ac836f-c3e5-42b1-9db4-69f796675b65)

The newest released `1.0.3` isn't tagged as the "latest" release for github. I think if you go to the release and click edit near the top right there should be a checkbox or button on the edit page to mark it as the "latest" release. The bundle build tool will always look for the one tagged "latest" so it's good to have that always as the newest one. 

I believe the version number on the prior release is part of the issue as well `1.02` I believe the build tool expects that the release version tag will always be a 3 digit semantic version value so it must have two dots, `1.0.2` would be valid, but `1.02`  is not.

joepardue commented:
FoamyGuy, I think I did some work while you were commenting, so I'm not sure where we are at the moment. I submitted a more comprehensive README file and broke something, so let me look into that. I really do appreciate your looking at this. I'm slowly figuring it out.

FoamyGuy reviewed (approved):
Aweomse! Thank you @joepardue

joepardue commented:
I really appreciate your patience on this. Hopefully everything will now
stick.

On Thu, May 15, 2025 at 8:44 PM foamyguy ***@***.***> wrote:

> Merged #254
> <https://github.com/adafruit/CircuitPython_Community_Bundle/pull/254>
> into main.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/adafruit/CircuitPython_Community_Bundle/pull/254#event-17680943355>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGJBELQGLBQ2EBMOWEIPEV326UYGVAVCNFSM6AAAAAB4W5TGQCVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXGY4DAOJUGMZTKNI>
> .
> You are receiving this because you were mentioned.Message ID:
> <adafruit/CircuitPython_Community_Bundle/pull/254/issue_event/17680943355@
> github.com>
>
